### DISCUSSION TITLE
mod won't load ( ó ︿ò)

### DISCUSSION THREAD

**Discussion by eksynn** [thatskymod/Sky-CotL-Scripts#161] [Q&A] [answered]
TSM: Starting initialization...
TSM: Hardware ID mismatch
TSM: Exception during initialization: DLL integrity check failed
TSM: Core not initialized, returning

**Comment by Slingexe** [comment, marked as answer]
Yeah because of people selling Trinity's mod to Chinese players for money you now have to download the mod from his app called [ThatSkyApp](https://github.com/XeTrinityz/ThatSkyApp/releases/tag/1.8.2)

**Reply by eksynn** [reply]
i already had downloaded it from there, but i downloaded again just in case. as it turns out, the updater on TSA didn't update TSA xD
thanks for the help! it's fixed now

**Reply by Slingexe** [reply]
Lol that's interesting.

**Comment by thatskymod** [comment]
Correct, @Slingexe! The mod must now be downloaded through what is essentially a mod manager, [ThatSkyApp](https://github.com/XeTrinityz/ThatSkyApp/releases/tag/1.8.4). Only then will it pass the integrity check and initialize properly.
